### ISSUE TITLE
Game not launching on MacOS 10.13.6 High Sierra

### ISSUE THREAD

**Issue by cirru05** [Quaver/Quaver#3610]
Title disclaimer: I do not know if it works for newer versions or not, but the update has been out for 4 days now, and it doesn't seem like anybody else is having the problem.

**Describe the bug**
I am running an older version of MacOS, and whenever I click play in the Steam client the game refuses to start up. It does not start up and crash afterwards, it just simply never launches.
But launching the game from the command line (<steam_library>/steamapps/common/Quaver/Quaver) gives the following error:
```
Kisss-iMac:Quaver boldizsar$ ./Quaver
Unhandled exception. System.DllNotFoundException: Unable to load shared library 'System.Native' or one of its dependencies. In order to help diagnose loading problems, consider setting the DYLD_PRINT_LIBRARIES environment variable: dlopen(libSystem.Native, 1): image not found
   at Interop.Sys.Stat(Byte& path, FileStatus& output)
   at Interop.Sys.Stat(ReadOnlySpan`1 path, FileStatus& output)
   at System.IO.FileSystem.CreateDirectory(String fullPath)
   at System.IO.Directory.CreateDirectory(String path)
   at Wobble.Logging.Logger.Initialize()
   at Quaver.Program.Run()
   at Quaver.Program.Main(String[] args)
Abort trap: 6
```

After taking the error's advice I run it again, but set the environmental var to the desired value and got even more information (`DYLD_PRINT_LIBRARIES environment variable: dlopen(libSystem.Native, 1): image not found`): https://pastebin.com/jA45zf8z
(it is too long of a string to be pasted here)
hmmm I think it doesn't like my System.Native.dylib smh

**To Reproduce**
1. Have a Mac, preferably on MacOS 10.13.6 High Sierra
2. Launch the game
3. See error

**Expected behaviour**
The game starts up

**Screenshots**
uhhhh... no, instead here is link to a cool little video: [how to use md links?](https://youtu.be/Rk1MYMPDx3s)

**Desktop:**
 - OS: MacOS 10.13.6 (also known as High Sierra)
 - Browser: Safari, Chrome
 - Version: 0.32.0

**Additional context**
nope, just try the play the game

**Comment by AiAe** [comment]
We've tried to debug why this happens, since nothing in that regard was changed. 
Also verified that the game was targeted for dotnet 3.1 (which before when was compiled under newer this issue was accruing).

Since I or no one from our staff has old Mac, can I ask you to try to install from brew or stand alone dotnet 3.1 and then try booting the game from `quaver.sh` or just from Steam again?

**Comment by cirru05** [comment]
Thank you @AiAe for the reply!
I already had dotnet 5.0 installed, but tried to install 3.1 on the top of it. This resulted in some conflict and the `dotnet --version` command still showed me that the 5.0 is installed.
I got rid of the 5.0 and installed the 3.1 but it just 'unexpectedly quits' even when I just simply run the `dotnet --version` command to check if it is installed. 
I tried to run the `quaver.sh` script in the game folder but the terminal greeted me with an error and an Apple 'unexpected quit' message appeared as well. The Apple one was regarded to the process `dotnet` crashing.
So now I have dotnet 3.1 which does not seem to work (Microsoft programs doing Microsoft things).
If you need it here is the crash log for dotnet and the one from quaver.sh as well: https://pastebin.com/yYD7Tr8b

**Comment by cirru05** [comment]
Greetings, devs and contributors!
It has been a while since this issue was last active, so I am asking if any further help is needed or not.
If I can do anything that helps fixing this issue, than feel free to contact me.
Or, if because of this issue the game will not be playable any more on older Macs, do not be afraid to tell me that as well.
Looking forward to any and every response.
Yours faithfully,
Me

**Comment by AiAe** [comment]
Sadly there is no further investigation going on, because none of us can test it or know why exactly it happens.
Other than suggesting for you to try compiling the game yourself and see if that runs fine, but that also means you wont be able to play online.

**Comment by cirru05** [comment]
Thank you for your reply.
I tried to compile it, it threw similar errors.
Wishing good luck regarding further developments of the game.
Until next time,
Bye & have fun!

**Comment by cirru05** [comment]
Thank you for your reply.
I tried to compile it, it threw similar errors.
Wishing good luck regarding further developments of the game.
Until next time,
Bye & have fun!

**Comment by AiAe** [comment]
Thats really weird, if you understand git I can suggest u try few versions/history back before last stuff was added. 
If that fixes it.

**Comment by AiAe** [comment]
You can revert back to here https://github.com/Quaver/Quaver/tree/1e1aa2fe0f647c94b9dbeba872f49e9e18f212c2 and test again

**Comment by AiAe** [comment]
> Ok, I am up for it. Which one do you recommend building?

Probably didn't see but i gave u history link from where u can test

**Comment by cirru05** [comment]
Hello there!
I have seen the history link, and it worked, thank you for that one.
But I have a question: is it possible to log in to non-steam version of the game and submit scores, or is it disabled because of the risk of the code being modified in order to cheat? (so can I play online w/ this build?)

**Comment by AiAe** [comment]
Yeah it is disabled, and thanks for trying that one I will see what exactly was changed so we can fix it to work again on older macos version.

**Comment by AiAe** [comment]
@cirru05 
Since you have it already setup can I ask you try and see how recent in history you can get the game to run? 
https://github.com/Quaver/Quaver/commits/ui-redesign 
So the history i linked you was from Jul 10 can you try the next few commits after it and see when it breaks.

**Comment by cirru05** [comment]
Hi!
It turned out that the .NET v3.1 framework's latest version for Mac is not too friendly to an older OS (it straight up crashes upon launching). So I installed an earlier one, and the game -  when building from source - started working again. I tried even the latest commit currently available, and it launched just like it normally would.
But when I tried to play in the Steam version, it resulted in me staring at the same errors I mentioned in the first post here. I also reinstalled it, to find out if it was only a problem with my current instance, but with no success, the problem stayed.
If I can help with anything else, I am happy to provide more info!
